### ISSUE TITLE
DOC: The PDF documentation does not look good.

### ISSUE THREAD

**Issue by charris** [numpy/numpy#19299]
* The parameter type following the `:` is put in  brackets `[...]` with the description and are hard to make out.
* There are unresolved references such as   \`(b-a)\` that are not references.
* Internal references don't go anywhere.
* There is an unrecognized character somewhere in the documentation, maybe generated.
* There seems to be a filename mixup between `index.ind` and `index.idx`.

Some of these problems are likely due to sphinx (4.0.1). The problems show up when running `make dist`.

**Comment by charris** [comment]
The undefined characters are because sphinx cannot handle byte strings in verbatim examples, tries to convert them to characters for pdf.
```
        Examples
        --------
        >>> np.random.default_rng().bytes(10)
        b'\xfeC\x9b\x86\x17\xf2\xa1\xafcp' # random
```
Bytes looks like a weird function anyway, as I don't believe 32 bit unicode is densely defined, yet the function converts 32 bit ints to '<U4' to bytes.

**Comment by rkern** [comment]
No, it's `<u4`, i.e. little-endian `uint32`s.

**Comment by charris** [comment]
Changing the example outputs for bytes to `b'0123456789'` allows the pdf docs to build, but that doesn't look very random :)

**Comment by charris** [comment]
> No, it's <u4, i.e. little-endian uint32s

OK.

**Comment by rkern** [comment]
It's a bug (or missing feature) in the PDF generator if the HTML output looks fine.

**Comment by charris** [comment]
It is a PDF problem, but the problem is in the generated latex code, so I suspect sphinx. Might be mklatex, I suppose, as that is the other thing that has probably changed, but sphinx has changed from `2.2.0` (NumPy 1.20.x) to `4.0.1` (NumPy 1.21.x).

**Comment by charris** [comment]
The other thing that has changed is the example output itself, from `b' eh\x85\x022SZ\xbf\xa4'` to `b'\xfeC\x9b\x86\x17\xf2\xa1\xafcp'`. I note that the pdf docs omit the `b` prefix. IPython also fails to render the first correctly when it is part of the docstring.

**Comment by melissawm** [comment]
Can [this be related](https://github.com/sphinx-doc/sphinx/issues/7735)?

**Comment by rkern** [comment]
Yes, we should try using a raw string for the docstring. Though I do wonder why the HTML would come out fine.

**Comment by melissawm** [comment]
@charris when you say internal links don't go anywhere, is it because they are unresolved references? If so, running `make dist` twice (the joys of LaTeX!) does the trick for me.

Just so that I can understand, 

> The parameter type following the `:` is put in brackets `[...]` with the description and are hard to make out.

Is this something that recently changed, or has is been like this for a while?

**Comment by charris** [comment]
@melissawm Oh, that looks like the same problem. So maybe all that is needed is doubling the `\`.

**Comment by rkern** [comment]
As long as it continues to look right in the HTML and `help()`. That already seems to be the case without the `r"""` or doubled `\`, which is what I find a little puzzling.

**Comment by charris** [comment]
>Is this something that recently changed, or has is been like this for a while?

@melissawm I don't know, but suspect it has been the case for a while. What I mean by not going anywhere is that they aren't links in the pdf document to other parts of the pdf document, which I think can be done. The doc build should already be running multiple passes, there is a script somewhere. . . The unresolved links are just emphasized, while the resolved links don't do anything. That could be a problem with the viewer, I suppose. There are also a couple of other complaints. We don't build the pdf docs in the testing, so I expect things just slip through. No doubt there is much that I am mistaken about at this point.

**Comment by charris** [comment]
@rkern 

The help, or at least the `whatever?` ipython version, do show problems. I'd need to double check the html to make sure it has them right, but that is my recollection.

**Comment by rkern** [comment]
Can you show me? I don't see those problems.

**Comment by rkern** [comment]
```
[~]
|2> np.__version__
'1.20.1'

[~]
|3> np.random.Generator.bytes?
Docstring:
bytes(length)

Return random bytes.

Parameters
----------
length : int
    Number of random bytes.

Returns
-------
out : str
    String of length `length`.

Examples
--------
>>> np.random.default_rng().bytes(10)
' eh\x85\x022SZ\xbf\xa4' #random
Type:      method_descriptor

[~]
|4> np.random.Generator.bytes.__doc__
"\n        bytes(length)\n\n        Return random bytes.\n\n        Parameters\n        ----------\n        length : int\n            Number of random bytes.\n\n        Returns\n        -------\n        out : str\n            String of length `length`.\n\n        Examples\n        --------\n        >>> np.random.default_rng().bytes(10)\n        ' eh\\x85\\x022SZ\\xbf\\xa4' #random\n\n        "

[~]
|5> help(np.random.Generator.bytes)
Help on method_descriptor:

bytes(...)
    bytes(length)
    
    Return random bytes.
    
    Parameters
    ----------
    length : int
        Number of random bytes.
    
    Returns
    -------
    out : str
        String of length `length`.
    
    Examples
    --------
    >>> np.random.default_rng().bytes(10)
    ' eh\x85\x022SZ\xbf\xa4' #random
```

**Comment by rkern** [comment]
Ah, 1.20.1 did have them doubled in the source. I thought I had a source checkout where they were not doubled. Never mind.

**Comment by charris** [comment]
@rkern 

The html was also broken, but not for the legacy version of bytes which had the escaped backslashes. What you show above is the same as the legacy, but that was changed in #18395, see https://tinyurl.com/wr3u5twc.

**Comment by charris** [comment]
Thanks @melissawm and @rkern I now know what went wrong with the build.

A small note on the new site layout, I find it hard to locate the function api documentation amidst all the explanatory material. Is there a way we could make that easier? An easily found index would help, maybe in the header bar? I've always appreciated Beazley's bare essentials approach :)

**Comment by melissawm** [comment]
@charris is this improved by #19756 ?
